Build failure: linuxManualConfig with ccacheStdenv

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

```nix
linuxManualConfig = prev'.linuxManualConfig.override {
  stdenv = final'.ccacheStdenv;
  buildPackages = final'.buildPackages // {
    stdenv = final'.ccacheStdenv;
  };
};
```

### Can Hydra reproduce this build failure?

Hydra doesn’t try to build the package.

### Link to Hydra build job

_No response_

### Relevant log output

```console
manual-config configurePhase buildRoot=/build/source/build pwd=/build/source
'/build/source/build/.config' -> '/nix/store/94pncs1zladi2zl6c9xzalsc5asa4qyy-kernel-configuration'
make[1]: Entering directory '/build/source/build'
  GEN     Makefile
  HOSTCC  scripts/basic/fixdep
  HOSTCC  scripts/kconfig/conf.o
  HOSTCC  scripts/kconfig/confdata.o
  HOSTCC  scripts/kconfig/expr.o
  LEX     scripts/kconfig/lexer.lex.c
  YACC    scripts/kconfig/parser.tab.[ch]
  HOSTCC  scripts/kconfig/lexer.lex.o
  HOSTCC  scripts/kconfig/menu.o
  HOSTCC  scripts/kconfig/parser.tab.o
  HOSTCC  scripts/kconfig/preprocess.o
  HOSTCC  scripts/kconfig/symbol.o
  HOSTCC  scripts/kconfig/util.o
  HOSTLD  scripts/kconfig/conf
scripts/Kconfig.include:40: C compiler '/nix/store/fq4n47ndbqi17d8g1fbn58vj5kzrw5bx-ccache-links-4.11.3/bin/ccache-links' not found
make[3]: *** [../scripts/kconfig/Makefile:85: oldconfig] Error 1
make[2]: *** [/build/source/Makefile:730: oldconfig] Error 2
make[1]: *** [/build/source/Makefile:248: __sub-make] Error 2
make[1]: Leaving directory '/build/source/build'
make: *** [Makefile:248: __sub-make] Error 2
```

### Additional context

evaluation warning: getExe: Package "ccache-links-4.11.3" does not have the meta.mainProgram attribute. We'll assume that the main program has the same name for now, but this behavior is deprecated, because it leads to surprising errors when the assumption does not hold. If the package has a main program, please set `meta.mainProgram` in its definition to make this warning go away. Otherwise, if the package does not have a main program, or if you don't control its definition, use getExe' to specify the name to the program, such as lib.getExe' foo "bar".

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.15.10-xanmod1, NixOS, 25.11 (Xantusia), 25.11.20250819.2007595`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.4`
 - nixpkgs: `/nix/store/pxnx7dbr0pz3qpgw2r315wdbb00afdw7-source`

### Notify maintainers

_No response_

### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/cc/ccache/package.nix
+++ b/pkgs/by-name/cc/ccache/package.nix
@@ -158,6 +158,10 @@ stdenv.mkDerivation (finalAttrs: {
               ln -s ${unwrappedCC}/$file $out/$file
             done
           '';
+
+        meta = {
+          inherit (unwrappedCC.meta) mainProgram;
+        };
       };
 
     updateScript = nix-update-script { };

